src/Illuminate/Database/Eloquent/Builder.php

```diff
@@ -1070,18 +1070,94 @@ public function without($relations)
     }
 
     /**
+     * Add subselect queries to average a column value of a relation.
+     *
+     * @param  mixed  $relation
+     * @param  string $column
+     * @return $this
+     */
+    public function withAvg($relation, $column)
+    {
+        return $this->withAggregate('avg', $relation, $column);
+    }
+
+    /**
      * Add subselect queries to count the relations.
      *
      * @param  mixed  $relations
      * @return $this
      */
     public function withCount($relations)
     {
+        $relations = is_array($relations) ? $relations : func_get_args();
+
+        return $this->withAggregate('count', $relations, '*');
+    }
+
+    /**
+     * Add subselect queries to get max column value of a relation.
+     *
+     * @param  mixed  $relation
+     * @param  string $column
+     * @return $this
+     */
+    public function withMax($relation, $column)
+    {
+        return $this->withAggregate('max', $relation, $column);
+    }
+
+    /**
+     * Add subselect queries to get min column value of a relation.
+     *
+     * @param  mixed  $relation
+     * @param  string $column
+     * @return $this
+     */
+    public function withMin($relation, $column)
+    {
+        return $this->withAggregate('min', $relation, $column);
+    }
+
+    /**
+     * Add subselect queries to sum a column value of a relation.
+     *
+     * @param  mixed  $relation
+     * @param  string $column
+     * @return $this
+     */
+    public function withSum($relation, $column)
+    {
+        return $this->withAggregate('sum', $relation, $column);
+    }
+
+    /**
+     * Add subselect queries to aggregate a column value of a relation.
+     *
+     * @param  string $aggregate
+     * @param  mixed  $relation
+     * @param  string $column
+     * @return $this
+     */
+    protected function withAggregate($aggregate, $relation, $column)
+    {
+        return $this->withRaw($relation, new Expression($aggregate.'('.$column.')'), $aggregate);
```

Comment:
We should probably wrap the column...
```php
$this->query->grammar->wrap($column)
```
And call it directly for the withCount...